src/ocean/core/SmartUnion.d

```diff
@@ -289,6 +289,43 @@ version ( UnitTest )
     import ocean.io.Stdout;
 }
 
+/*******************************************************************************
+
+    Helper mixin to support final switch over a smart union: it handles the
+    `case none:` in D2-build and `default:` in D1.
+
+*******************************************************************************/
+
+const handleInvalidCases = "case none: assert(false);" ~
+    // The D2 build makes sure all enum cases are covered so in fact we don't
+    // need a default. However, D1 emits a compiler warning if a switch
+    // has no default so we add a default here just to avoid the warning.
+    "version (D_Version2){} else {default: assert(false);}";
+
+///
+unittest
+{
+    union TestUnion
+    {
+        int a;
+        float b;
+    }
+    alias SmartUnion!(TestUnion) TestSmartUnion;
+
+    TestSmartUnion u;
+    u.a = 1;
+
+    with (TestSmartUnion.Active) final switch (u.active)
+    {
+        case a:
+            break;
+        case b:
+            break;
+
+        mixin(handleInvalidCases);
```

Comment:
This usage makes me wonder if it might make sense to add an alias to `handleInvalidCases` inside `SmartUnion!`. So you could do `mixin(u.handleInvalidCases);` or `mixin(TestSmartUnion.handleInvalidCases);`.

Reply:
Oh, that's clever!